docs/src/token.md

```diff
@@ -476,3 +476,139 @@ have a balance of zero to be closed.
+        ],
+        &spl_associated_token_account::id()
+    ).0
+}
+```
+
+
+Javascript equivalent:
+```
+import {PublicKey, PublicKeyNonce} from '@solana/web3.js';
+
+async function findDefaultTokenAddress(
+    walletAddress: Pubkey,
+    tokenMintAddress: Pubkey
+): Promise<PublicKeyNonce> {
+    return PublicKey.findProgramAddress(
+        [
+            walletAddress.toBuffer(),
+            TOKEN_PROGRAM_ID.toBuffer(),
+            tokenMintAddress.toBuffer(),
+        ],
+        SPL_ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID
+    );
+}
+```
+
+### Ancillary Token Accounts
+At any time ownership of an existing SPL Token account may be assigned to the
+user.  One way to accomplish this is with the
+`spl-token authorize <TOKEN_ADDRESS> owner <USER_ADDRESS>` command.
+
+By convention it is suggested that wallets roll up the balances from all token
+accounts of the same mint into a single balance for the user to shield the user
+from this complexity.
+
+### How to fetch and display token holdings
+The [getTokenAccountsByOwner](https://docs.solana.com/apps/jsonrpc-api#gettokenaccountsbyowner)
+JSON RPC method can be used to fetch all token accounts for a wallet address.
+
+### Default Token Account Funding
+Before the user can receive tokens their associated token account must be created
+on-chain, requiring a small amount of SOL to mark the account as rent-exempt.
+
+There's no restriction on who can create a user's associated token account.  It
+could either be created by the wallet on behalf of the user or funded by a 3rd
+party through an airdrop campaign.
+
+Ultimately the
+`spl_associated_token_account::create_associated_token_account()` instruction
+just needs to be executed by some party.
+
+### Associated Token Account Ownership
+The wallet should never use `TokenInstruction::SetAuthority` to set the
+`AccountOwner` authority of the associated token account to another address.
+
+#### Sample "Add Token" workflow
+The wallet could provide a UI that allows users to "add a token".  The user
+selects the kind of token, and is presented with information about how much SOL
+it will cost them.  Upon confirmation, the wallet sends a transaction with the
+`spl_associated_token_account::create_associated_token_account()` instruction
+
+#### Sample "Airdrop campaign" workflow
+For each recipient wallet addresses, send a transaction containing:
+1. `spl_associated_token_account::create_associated_token_account()` to create
+   the recipient's associated token account if necessary
+2. `TokenInstruction::Transfer` to complete the airdrop
+
+### Transferring Tokens Between Wallets
+The preferred method of transferring tokens between wallets is to transfer into
+associated token account of the recipient.
+
+The recipient must provide the main wallet address to the sender.  The sender
+then derives the associated token account for the recipient, and uses
+`TokenInstruction::Transfer` to complete the transfer.
```

Comment:
Not sure how prescriptive we need to be here but I think it would be good to add the following steps:

1. Find associated token address for destination wallet address
2. Fetch associated token account and check that it exists and is owned by the destination wallet address
3. If associated token account is valid, transfer tokens. Otherwise, display error.